utils/cr_demo/run_demo_aurora.py

```diff
@@ -0,0 +1,433 @@
+
+TEST_CMD = "sudo date"
+test_cmd_list = TEST_CMD.split(" ")
+print(f"Check sudo works by calling: {TEST_CMD}")
+subprocess.run(test_cmd_list, check=False)
+
+POOL_SIZE = "5T"
+POOL_SIZE_F5 = "3T"
+POOL_LABEL = "tank"
+CONT_LABEL = "bucket"
+# Number of seconds to wait for engines to start for 1 group setup.
+FORMAT_SLEEP_SEC = 35
+
+print("\nF1: Dangling pool")
+print("F2: Lost the majority of pool service replicas")
+print("F3: Orphan pool")
+print("F4: Inconsistent pool label between MS and PS")
+print("F5: Orphan pool shard")
+print("F6: Dangling pool map")
+print("F7: Orphan container")
+print("F8: Inconsistent container label between CS and container property")
+
+PARSER = argparse.ArgumentParser()
+PARSER.add_argument(
+    "-l", "--hostlist", required=True, help="List of hosts to run the demo")
+ARGS = vars(PARSER.parse_args())
+
+HOSTLIST = ARGS["hostlist"]
+
+print(f"\n1. Format storage on {HOSTLIST}.")
+format_storage(host_list=HOSTLIST)
+
+print(f"\nWait for {FORMAT_SLEEP_SEC} sec for format...")
+time.sleep(FORMAT_SLEEP_SEC)
+
+# Call dmg system query to obtain the IP address of necessary ranks.
+rank_to_ip = {}
+stdout = system_query(json=True)
+# Printing system query output helps, but the output will be long if there are many ranks.
+# print(f"dmg system query stdout = {stdout}")
+generated_yaml = yaml.safe_load(stdout)
+RANK_COUNT = 0
+JOINED_COUNT = 0
+for member in generated_yaml["response"]["members"]:
+    rank_to_ip[member["rank"]] = member["addr"].split(":")[0]
+    RANK_COUNT += 1
+    if member["state"] == "joined":
+        JOINED_COUNT += 1
+# Print the number of ranks and joined ranks as a reference.
+node_set = NodeSet(HOSTLIST)
+hostlist = list(node_set)
+print(f"\n{len(hostlist)} nodes; {RANK_COUNT} ranks; {JOINED_COUNT} joined")
+
+# Create rank to mount point map and host to ranks map for F2 and F5.
+# 1. scp daos_control.log from all nodes to here, where this script runs. scp the local
+# file as well. Add hostname to the end of the file name. The log contains rank and PID.
+# Number of nodes used for F2.
+NODE_COUNT = 2
+for i in range(NODE_COUNT):
+    scp_cmd_list = ["scp", f"{hostlist[i]}:/var/tmp/daos_testing/daos_control.log",
+                    f"/var/tmp/daos_testing/daos_control_{hostlist[i]}.log"]
+    subprocess.run(scp_cmd_list, check=False)
+
+# 2. Determine the rank to PID mapping from the control logs. In addition, determine the
+# host to ranks mapping for creating the pool. We need to know the four ranks for the
+# first two nodes. We'll use many nodes in Aurora, but only two nodes for F2.
+rank_to_pid = {}
+host_to_ranks = defaultdict(list)
+SEARCH_STR = r"DAOS I/O Engine.*process (\d+) started on rank (\d+)"
+for i in range(NODE_COUNT):
+    with open(
+        f"/var/tmp/daos_testing/daos_control_{hostlist[i]}.log", "r",
+            encoding="utf-8") as file:
+        for line in file:
+            match = re.findall(SEARCH_STR, line)
+            if match:
```

Comment:
No need to change, but FYI that `re.MULTILINE` is useful with `findall` for this purpose. Something like
```
matches = re.findall(search_str, output, re.MULTILINE)
for match in matches:
    # process
```

Reply:
Thanks for the comment. I'd like to fix it, but Aurora isn't available, so I'll leave it for now.